lib/TransactionRunner.js

```diff
@@ -528,102 +528,97 @@ Not performing HTTP request for '${transaction.name}'.\
 
   validateTransaction(test, transaction, callback) {
     logger.debug('Validating HTTP transaction by Gavel.js');
-    logger.debug('Determining whether HTTP transaction is valid (getting boolean verdict)');
-    gavel.isValid(transaction.real, transaction.expected, 'response', (isValidError, isValid) => {
-      if (isValidError) {
-        logger.debug('Gavel.js validation errored:', isValidError);
-        this.emitError(isValidError, test);
+
+    logger.debug('Validating HTTP transaction (getting verbose validation result)');
+    gavel.validate(transaction.real, transaction.expected, 'response', (validateError, gavelResult) => {
+      if (validateError) {
+        logger.debug('Gavel.js validation errored:', validateError);
+        this.emitError(validateError, test);
       }
 
       test.title = transaction.id;
       test.actual = transaction.real;
       test.expected = transaction.expected;
       test.request = transaction.request;
 
+      const isValid = gavelResult ? gavelResult.isValid : false;
```

Comment:
There are occasions when `gavelResult` is `null`. I believe the only scenario for this to happen is when `validateError` exists. It appears validation error doesn't terminate the Dredd process (which is expected, as there are multiple transactions to be validated, and invalid transaction is _expected_ result), but neither it stops the logic flow in this block.

@honzajavorek, please, do you think it's a correct behavior for a single transaction logic to continue if `validateError` exists?

Reply:
Yes, this is correct. The (original) code isn't beautiful, but it works as it's supposed to work. This seems right.